static/js/lib/learning_resources_test.js

```diff
@@ -94,3 +105,245 @@ describe("Course utils", () => {
+      expected
+    )} for filter ${filter} and ending=false`, () => {
+      const parsedStartDate = availabilityFilterToMoment(filter, false)
+      assert.equal(
+        parsedStartDate ? parsedStartDate.format(DATE_FORMAT) : parsedStartDate,
+        expected ? `${expected}00:00:00Z` : null
+      )
+      const parsedEndDate = availabilityFilterToMoment(filter, true)
+      assert.equal(
+        parsedEndDate ? parsedEndDate.format(DATE_FORMAT) : parsedEndDate,
+        expected ? `${expected}23:59:59Z` : null
+      )
+    })
+  })
+
+  //
+  ;[
+    [
+      "2019-08-01T00:00:00Z",
+      "2019-08-31T00:00:00Z",
+      ["availableNow"],
+      true,
+      AVAILABILITY_MAPPING.availableNow.label
+    ],
+    [
+      "2019-08-02T00:00:00Z",
+      "2019-08-31T00:00:00Z",
+      ["nextWeek"],
+      false,
+      AVAILABILITY_MAPPING.availableNow.label
+    ],
+    [
+      "2019-08-02T00:00:00Z",
+      "2019-08-31T00:00:00Z",
+      ["availableNow", "nextWeek"],
+      true,
+      AVAILABILITY_MAPPING.availableNow.label
+    ],
+    [
+      "2019-09-02T00:00:00Z",
+      "2019-10-31T00:00:00Z",
+      ["nextWeek"],
+      true,
+      AVAILABILITY_MAPPING.nextWeek.label
+    ],
+    [
+      "2019-08-01T00:00:00Z",
+      null,
+      ["availableNow"],
+      true,
+      AVAILABILITY_MAPPING.availableNow.label
+    ],
+    [
+      null,
+      "2019-08-31T00:00:00Z",
+      ["availableNow"],
+      true,
+      AVAILABILITY_MAPPING.availableNow.label
+    ],
+    [
+      null,
+      "2019-08-31T00:00:00Z",
+      ["nextWeek"],
+      false,
+      AVAILABILITY_MAPPING.availableNow.label
+    ],
+    [
+      null,
+      null,
+      ["availableNow"],
+      true,
+      AVAILABILITY_MAPPING.availableNow.label
+    ],
+    [null, null, ["nextWeek"], false, AVAILABILITY_MAPPING.availableNow.label],
+    [
+      "2019-09-02T00:00:00Z",
+      "2019-09-30T00:00:00Z",
+      ["availableNow"],
+      false,
+      AVAILABILITY_MAPPING.nextWeek.label
+    ],
+    [
+      "2019-09-02T00:00:00Z",
+      null,
+      ["availableNow"],
+      false,
+      AVAILABILITY_MAPPING.nextWeek.label
+    ],
+    [
+      null,
+      "2019-09-31T00:00:00Z",
+      ["availableNow"],
+      true,
+      AVAILABILITY_MAPPING.availableNow.label
+    ],
+    [
+      "2019-09-02T00:00:00Z",
+      "2019-09-30T00:00:00Z",
+      ["availableNow", "nextWeek"],
+      true,
+      AVAILABILITY_MAPPING.nextWeek.label
+    ],
+    [
+      "2019-09-02T00:00:00Z",
+      null,
+      ["availableNow", "nextWeek"],
+      true,
+      AVAILABILITY_MAPPING.nextWeek.label
+    ],
+    [
+      null,
+      "2019-09-30T00:00:00Z",
+      ["availableNow", "nextWeek"],
+      true,
+      AVAILABILITY_MAPPING.availableNow.label
+    ],
+    [
+      "2020-08-02T00:00:00Z",
+      null,
+      ["nextYear"],
+      true,
+      AVAILABILITY_MAPPING.nextYear.label
+    ],
+    ["2030-08-02T00:00:00Z", null, ["nextYear"], false, null]
```

Comment:
just curious, what does it mean for the start or end date to be null? what circumstances will that be the case?

Reply:
The quality of the imported course run data is all over the place, some of them are missing start dates and/or end dates.